src/base.jl

```diff
@@ -23,25 +23,28 @@ function issquare(A::AbstractMatrix)
     return m == n
 end
 
-abstract type AbstractSquareKronecker <: AbstractKroneckerProduct end
+"""
+    issquare(K::AbstractKroneckerProduct)
 
-# general Kronecker product between two square matrices
-struct SquareKroneckerProduct{T, TA<:AbstractMatrix, TB<:AbstractMatrix} <: AbstractSquareKronecker
-    A::TA
-    B::TB
-    function SquareKroneckerProduct(A::AbstractMatrix{T}, B::AbstractMatrix{V}) where {T, V}
-        if issquare(A) && issquare(B)
-            return new{promote_type(T, V), typeof(A), typeof(B)}(A, B)
-        else
-            throw(DimensionMismatch(
-                "SquareKroneckerProduct is only for when all matrices are square",
-            ))
-        end
-    end
+Checks if all matrices of a Kronecker product are square.
+"""
+function issquare(K::AbstractKroneckerProduct)
+    return issquare(K.A) && issquare(K.B)
+end
+
+function squarecheck(K::AbstractKroneckerProduct)
+    issquare(K) || throw(DimensionMismatch("kronecker system is not composed of two square matrices: $size(K.A) and $size(K.B)"))
```

Comment:
Exceeds 92 char limit, so would be good to shift the error message onto the next line, maybe
```julia


issquare(K) || throw(DimensionMismatch(
    "kronecker system is not composed of two square matrices: $size(K.A) and $size(K.B)"))
```